### DISCUSSION TITLE
Getting maxLength

### DISCUSSION THREAD

**Discussion by smcardle** [colinhacks/zod#1762] [Q&A] [answered]
Hi.
I want to check if a property of a schema has a maxLength.
If I try the following I get an error because unwrap() either does NOT exist because it's not optional OR because .maxLength does not exist on on it if it IS optional
ERROR IS - Property 'maxLength' does not exist on type 'ZodOptional<ZodString>'.

```ts
const maxLen = PatientDetails.shape.familyName.isOptional()
    ? PatientDetails.shape.familyName.unwrap().maxLength || undefined
    : PatientDetails.shape.familyName.maxLength || undefined
```
I just want to reliable check if it has a maxLength or not no matter if it optional or not.

Anybody know how to do this ??? I am sing TypeScript if that make any difference i.e the above failure.

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

```ts
function getMaxLength ( schema: z.ZodTypeAny ): number | undefined {
    const hasMaxLength = z.object( { maxLength: z.number() } ).safeParse( schema )
    if ( hasMaxLength.success ) return hasMaxLength.data.maxLength

    if ( 'unwrap' in schema && typeof schema.unwrap === 'function' )
        return getMaxLength( schema.unwrap() )
}

console.log( getMaxLength( z.string().max( 20 ) ) ) // 20
console.log( getMaxLength( z.string().max( 20 ).optional() ) ) // 20
console.log( getMaxLength( z.string().max( 20 ).optional().nullable() ) ) // 20
console.log( getMaxLength( z.string().max( 20 ).nullish() ) ) // 20
```

**Reply by smcardle** [reply]
That's prettymuch what I was looking for yes.

Thanks

**Reply by JacobWeisenburger** [reply]
glad I could help
